### PR TITLE
Added method to do ZIP with array of arrays.

### PR DESCRIPTION
**Pull request by isaacroldan**
#### :tophat: What? Why?

There is already a `zipWith` method but i think this one is more useful.
Zip an array of arrays.
this:

``` objc
@[@[@1, @2, @3], @[@4, @5, @6], @[@7, @8, @9]]
```

transforms into this:

``` objc
@[@[@1, @4, @7], @[@2, @5, @9], @[@3, @6, @9]]
```

:warning: all the arrays should have the same lenght

### CODE REVIEW

**Comment by isaacroldan** [comment]
@masylum @sergigracia

**Comment by mariabernis** [comment]
:+1:

**Comment by masylum** [comment]
I don't see tests nor a PR to the original repo!

**Comment by isaacroldan** [comment]
The PR to the original Repo is this one:
https://github.com/robb/Underscore.m/pull/60

i'm working in tests, but they use a very old cocoapods version and i can't run them :(

File: Underscore/USArrayWrapper.m [hunk 1]
```diff
@@ -243,6 +243,19 @@ - (USArrayWrapper *)shuffle
     };
 }
 
+- (USArrayWrapper *)zip
+{
+    int lenght = [self.first count];
+    NSMutableArray *finalMutableArray = [NSMutableArray arrayWithCapacity:lenght*self.array.count];
```
Comment (masylum) [inline Underscore/USArrayWrapper.m]: typo `s/lenght/length`

Comment (masylum) [inline Underscore/USArrayWrapper.m]: and :art: spaces between operators

**Comment by isaacroldan** [comment]
I'll add tests in another moment, I can't even compile their test project.

File: Underscore/USArrayWrapper.m [hunk 2]
```diff
@@ -243,6 +243,19 @@ - (USArrayWrapper *)shuffle
     };
 }
 
+- (USArrayWrapper *)zip
+{
+    int length = [self.first count];
+    NSMutableArray *finalMutableArray = [NSMutableArray arrayWithCapacity:lenght * self.array.count];
```
Comment (masylum) [inline Underscore/USArrayWrapper.m]: typo is still here....

Comment (isaacroldan) [inline Underscore/USArrayWrapper.m]: agh, stupid me, sorry, i'll fix it

**Comment by sergigracia** [comment]
:+1:

**Pull request merged**